docs/standard/microservices-architecture/multi-container-microservice-net-applications/implementing-api-gateways-with-ocelot.md

```diff
@@ -0,0 +1,579 @@
+    - ASPNETCORE_URLS=http://0.0.0.0:80
+    - ConnectionString=YOUR_VALUE
+    - ... Other Environment Variables
+  ports:
+    - "5101:80"   # Important: In a production environment you should remove the external port (5101) kept here for microservice debugging purposes. 
+                  # The API Gateway redirects and access through the internal port (80).
+```
+
+You can see how in the docker-compose.override.yml configuration the internal port for the Catalog container is port 80, but the port for external access is 5101. But this port shouldn’t be used by the application when using an API Gateway, only to debug run and test just the Catalog microservice.
+
+Normally, you won’t be deploying with docker-compose into a production environment because the right production deployment environment for microservices is an orchestrator like Kubernetes or Service Fabric and when deploying to those environments you will use different configuration files where you won’t publish directly any external port for the microservices but you will always use the reverse proxy from the API Gateway.
+
+Let’s run the catalog microservice in your local Docker host either by running the full eShopOnContainers solution from Visual Studio (it’ll run all the services in the docker-compose files) or just starting the Catalog microservice with the following docker-compose command in CMD or PowerShell positioned at the folder where the docker-compose.yml and docker-compose.override.yml are placed.
+
+```csharp
+docker-compose run --service-ports catalog.api
+```
+
+This command will only run the catalog.api service container plus its dependencies which are specified in the docker-compose.yml.
+In this case, the dependencies are the SQL Server container and RabbitMQ container.
+
+Then you could directly access the Catalog microservice and see its methods through the Swagger UI accessing directly through that “external” port, in this case http://localhost:5101 
+
+![](./media/image31.png)
+
+**Figure 8-30.** Testing the Catalog microservice with its Swagger UI
+
+At this point, you could set a breakpoint in C# code in Visual Studio, test the microservice with the methods exposed in Swagger UI, etc. and finally clean-up everything with the “docker-compose down” command.
+
+However, this direct-access communication to the microservice, in this case through the external port 5101, is precisely what you want to avoid in your application. And you can avoid that by setting the additional level of indirection of the API Gateway, Ocelot, in this case. That way, the client app won’t directly access the microservice.
+
+
+## Implementing your API Gateways with Ocelot
+
+Ocelot is basically a set of middlewares that you can apply in a specific order.
+
+Ocelot is designed to work with ASP.NET Core only and it targets netstandard2.0 so it can be used anywhere .NET Standard 2.0 is supported, including .NET Core 2.0 runtime and .NET Framework 4.6.1 runtime and up.
```

Comment:
Break the sentence in the middle.

inline code the TFM

Reply:
👍